File: Sources/Sample/AutoGenerateJavaCodes/foregroundJni/CSingletonTest.java

package org.elastos.Elastos.AutoGenerateJavaCodes;

import java.util.Calendar;
import java.util.Random;

//TODO : Modify the package and the final user will call this class' method.
public class CSingletonTest {
    private static final String CLASS_ID_PREFIX = "org.elastos.Elastos.AutoGenerateJavaCodes.CSingletonTest.";
    private String mClassId;

    private native void native_CSingletonTest();
    private native void native_CSingletonTest_Destroy();

    private static class CSingletonTestHelper {
        private static final CSingletonTest INSTANCE = new CSingletonTest();
    }

    public static CSingletonTest getInstance() {
        return CSingletonTestHelper.INSTANCE;
    }

    private CSingletonTest() {
        Random rand = new Random();
        mClassId = CLASS_ID_PREFIX + rand.nextInt() + Calendar.getInstance().getTimeInMillis();
        native_CSingletonTest();
        //TODO: Add your code here
    }

    // invoke destroy to release car object
    public void destroy() {
        native_CSingletonTest_Destroy();
    }

    public String getClassId() {
        return mClassId;
    }


    public void SetInt2(
        /* [in] */ int value)
    {
        native_CSingletonTest_ITestCar2_SetInt2(value);
    }

    private native void native_CSingletonTest_ITestCar2_SetInt2(int value);

    public String Update(
        /* [in] */ String value1,
        /* [in] */ String value2,
        /* [in] */ String value3)
    {
        return native_CSingletonTest_ITestCar2_Update(value1, value2, value3);
    }

    private native String native_CSingletonTest_ITestCar2_Update(String value1, String value2, String value3);

    public void Test1(
        /* [in] */ String name,
        /* [in] */ int[] array)
    {
        native_CSingletonTest_ITestCar2_Test1(name, array);
    }

    private native void native_CSingletonTest_ITestCar2_Test1(String name, int[] array);

    public void Test2(
        /* [in] */ String[] strArray,
        /* [in] */ int[] intArray)
    {
        native_CSingletonTest_ITestCar2_Test2(strArray, intArray);
    }

    private native void native_CSingletonTest_ITestCar2_Test2(String[] strArray, int[] intArray);

    public String[] Test3(
        /* [in] */ int i)
    {
        return native_CSingletonTest_ITestCar2_Test3(i);
    }

    private native String[] native_CSingletonTest_ITestCar2_Test3(int i);

    public void Test4()
    {
        native_CSingletonTest_ITestCar2_Test4();
    }

    private native void native_CSingletonTest_ITestCar2_Test4();

    public String Test5()
    {
        return native_CSingletonTest_ITestCar2_Test5();
    }

    private native String native_CSingletonTest_ITestCar2_Test5();

    public float[] Test6(
        /* [in] */ int i,
        /* [in] */ String str)
    {
        return native_CSingletonTest_ITestCar2_Test6(i, str);
    }

    private native float[] native_CSingletonTest_ITestCar2_Test6(int i, String str);


}

File: Sources/Sample/AutoGenerateJavaCodes/foregroundJni/CSingletonTest.cpp

#include "CSingletonTest.h"



CAR_SINGLETON_IMPL(CSingletonTest)

CAR_INTERFACE_IMPL(CSingletonTest, Singleton, ITestCar2, IJavaInterface);

ECode CSingletonTest::SetInt2(
    /* [in] */ Int32 value)
{
    // TODO: Add your code here
    return E_NOT_IMPLEMENTED;
}

ECode CSingletonTest::Update(
    /* [in] */ const String& value1,
    /* [in] */ const String& value2,
    /* [in] */ const String& value3,
    /* [out] */ String * pValue)
{
    // TODO: Add your code here
    return E_NOT_IMPLEMENTED;
}

ECode CSingletonTest::Test1(
    /* [in] */ const String& name,
    /* [in] */ const ArrayOf<Int32> & array)
{
    // TODO: Add your code here
    return E_NOT_IMPLEMENTED;
}

ECode CSingletonTest::Test2(
    /* [in] */ const ArrayOf<String> & strArray,
    /* [in] */ const ArrayOf<Int32> & intArray)
{
    // TODO: Add your code here
    return E_NOT_IMPLEMENTED;
}

ECode CSingletonTest::Test3(
    /* [in] */ Int32 i,
    /* [out, callee] */ ArrayOf<String> ** ppStrArray)
{
    // TODO: Add your code here
    return E_NOT_IMPLEMENTED;
}

ECode CSingletonTest::Test4()
{
    // TODO: Add your code here
    return E_NOT_IMPLEMENTED;
}

ECode CSingletonTest::Test5(
    /* [out] */ String * pStr)
{
    // TODO: Add your code here
    return E_NOT_IMPLEMENTED;
}

ECode CSingletonTest::Test6(
    /* [in] */ Int32 i,
    /* [in] */ const String& str,
    /* [out, callee] */ ArrayOf<Float> ** ppFloatArray)
{
    // TODO: Add your code here
    return E_NOT_IMPLEMENTED;
}

ECode CSingletonTest::JavaInit(
    /* [in] */ Handle64 jvm)
{
    mJvm = (JavaVM*)jvm;
    assert(mJvm != NULL);

    return NOERROR;
}

ECode CSingletonTest::constructor()
{
    // TODO: Add your code here
    return E_NOT_IMPLEMENTED;
}

JNIEnv* CSingletonTest::GetEnv()
{
    JNIEnv* env;
    assert(mJvm != NULL);
    mJvm->AttachCurrentThread(&env, NULL);
    return env;
}

void CSingletonTest::Detach()
{
    assert(mJvm != NULL);
    mJvm->DetachCurrentThread();
}

